Added Service Statuses (#157)
DefangLabs/defang-docs: docs/concepts/services.md

:::info
We plan to add support for other types of services in the future, including serverless functions.
:::

### Status
Below is an overview of the possible statuses for a service in Defang. 

| Status | Details |
|-|-|
| BUILD_QUEUED | The build is being queued. | 

> | BUILD_PROVISIONING | The build is provisioning. | 

Reviewer: ```suggestion
| BUILD_PROVISIONING | The container orchestrator is provisioning the necessary resources for building your service's image. | 
```

> | BUILD_PENDING | The build is pending. | 

Reviewer: ```suggestion
| BUILD_PENDING | The necessary resources to build your service have been provisioned but the build has not yet been initiated. | 
```

> | BUILD_ACTIVATING | The build is activating. | 

Reviewer: ```suggestion
| BUILD_ACTIVATING | The container orchestrator is pulling the build container's image and creating the build container. | 
```

> | BUILD_RUNNING | The build is running. | 

Reviewer: ```suggestion
| BUILD_RUNNING | The container which builds your service's image is successfully running . | 
```

> | BUILD_STOPPING | The build is stopping. | 

Reviewer: ```suggestion
| BUILD_STOPPING | The container orchestrator has sent a termination signal to the build container and is waiting for the build process to stop. | 
```

> | BUILD_FAILED | The build failed. | 

Reviewer: ```suggestion
| BUILD_FAILED | The build container exited with a non-zero status code. | 
```

> | UPDATE_QUEUED | An update is being queued. | 

Reviewer: ```suggestion
| UPDATE_QUEUED | The service update has been received and is now in the queue. | 
```

> | SERVICE_DEPLOYMENT_PENDING | The deployment is pending. | 

Reviewer: ```suggestion
| SERVICE_DEPLOYMENT_PENDING | The necessary resources to run your service have been provisioned but the service has not yet been initiated. | 
```

> | SERVICE_DEPLOYMENT_COMPLETED | The deployment completed. | 

Reviewer: ```suggestion
| SERVICE_DEPLOYMENT_COMPLETED | Your service has been deployed and is healthy. | 
```

> | SERVICE_DEPLOYMENT_FAILED | The deployment failed. | 

Reviewer: ```suggestion
| [SERVICE_DEPLOYMENT_FAILED] | Your service could not be deployed. | 
```

| BUILD_QUEUED | The service update has been received and is now in the queue for its image to be built. | 
| BUILD_PROVISIONING | The container orchestrator is provisioning the necessary resources for building your service's image. | 
| BUILD_PENDING | The necessary resources to build your service have been provisioned but the build has not yet been initiated. | 
| BUILD_ACTIVATING | The container orchestrator is pulling the build container's image and creating the build container. | 
| BUILD_RUNNING | The container which builds your service's image is successfully running. | 
| BUILD_STOPPING | The container orchestrator has sent a termination signal to the build container and is waiting for the build process to stop. | 
| BUILD_FAILED | The build container exited with a non-zero status code. | 
| UPDATE_QUEUED | The service update has been received and is now in the queue. | 
| SERVICE_DEPLOYMENT_PENDING | The necessary resources to run your service have been provisioned but the service has not yet been initiated. | 
| SERVICE_DEPLOYMENT_COMPLETED | Your service has been deployed and is healthy. | 
| SERVICE_DEPLOYMENT_FAILED | Your service could not be deployed. | 

:::tip

> You can run the `defang compose ps` command to view a service's status.

Reviewer: nitpick, but `defang compose ps` is project-wide, so there may be more than one service.

```suggestion
You can run the `defang compose ps` command to view the status of your services.
```
